src/operator/split.spec.ts

```diff
@@ -0,0 +1,75 @@
+// operator/split.spec.ts
+
+/*
+ * Copyright (c) 2021 Check Digit, LLC
+ *
+ * This code is licensed under the MIT license (see LICENSE.txt for details).
+ */
+
+import * as assert from 'assert';
+
+import { from, pipeline, split, toArray } from '../index';
+
+describe('split', () => {
+  it('works with simple string values', async () => {
+    assert.deepStrictEqual(await pipeline(from(['a\nb\nc', 'd', 'e']), split('\n'), toArray), ['a', 'b', 'cde']);
+
+    assert.deepStrictEqual(await pipeline(from(['a\nb', '\nc', '\nd']), split('\n'), toArray), ['a', 'b', 'c', 'd']);
+  });
+
+  it('compatible with native split implementation', async () => {
+    async function check(value: string | string[], separator: string, limit?: number) {
+      if (typeof value === 'string') {
+        assert.deepStrictEqual(await toArray(split(separator, limit)(from([value]))), value.split(separator, limit));
+      } else {
+        assert.deepStrictEqual(
+          await toArray(split(separator, limit)(from(value))),
+          value.join('').split(separator, limit)
+        );
+      }
+    }
+    await check(['X', ''], '');
+    await check('X', '');
+    await check('X', 'X');
+    await check(['', '', ''], '');
+    await check('X', 'Y');
+    await check(['This is\na buffer than', ' needs to be split\ninto multiple lines\n'], '', 30);
+    await check('This is\na buffer than needs to be split\ninto multiple lines\n', '');
+    await check('This is\na buffer than needs to be split\ninto multiple lines\n', '', 15);
+    await check('This is\na buffer than needs to be split\ninto multiple lines\n', 'X');
+    await check('This is\na buffer than needs to be split\ninto multiple lines\n', '\n', 25);
+  });
+
+  it('supports limit', async () => {
+    assert.deepStrictEqual(await pipeline(from(['a\nb\nc', 'd', 'e']), split('\n', 0), toArray), []);
+    assert.deepStrictEqual(await pipeline(from([]), split('\n', 1), toArray), []);
```

Comment:
Does:
assert.deepStrictEqual(await pipeline(from['']), split('\n'), toArray), ['']);
work?

Reply:
diabolical, good catch.  fixed.

Reply:
btw while investigating this, discovered we don't handle edge cases like negative and non-integer values for limit.  Created another issue for this.